geomstats/geometry/special_orthogonal.py

```diff
@@ -105,12 +106,12 @@ def belongs(self, point, point_type=None):
             point_type = self.default_point_type
 
         if point_type == 'vector':
-            point = gs.to_ndarray(point, to_ndim=2)
-            n_points, vec_dim = point.shape
+            vec_dim = point.shape[-1]
             belongs = vec_dim == self.dimension
+            if point.ndim == 2:
+                belongs = gs.tile([belongs], (point.shape[0],))
             belongs = gs.to_ndarray(belongs, to_ndim=1)
             belongs = gs.to_ndarray(belongs, to_ndim=2, axis=1)
```

Comment:
Shouldn't this one be removed to output a 1d array?

Reply:
Yes! But so far I have only modified belongs and random_uniform for Euclidean and Minkowski (and fixed the issues that were appearing in the other files). I will move on and modify belongs and random on the other files now.